NoSuchMethodException in Resteasy-reactive when running in native mode

### Describe the bug

I have a sample of code, which runs successfully with Quarkus 2.7(including 2.7.6), but fails with `NoSuchMethodException` when running with Quarkus 2.8 and beyond in native mode. 
The method in question has a signature like that:
```
    @POST
    @Path("/upload")
    public Uni<String> upload(File body)
```

### Expected behavior

The code should either work or fail with errors/warning, which describe, what is happening.

### Actual behavior

An error:
```
11:41:13,380 INFO  [app] 11:41:12,362 HTTP Request to /file/upload failed, error id: d4b4d067-51d9-489f-aa9d-4a7808e36266-1: java.lang.RuntimeException: java.lang.NoSuchMethodException: io.quarkus.ts.http.restclient.reactive.files.FileResource.upload(java.io.File)
11:41:13,380 INFO  [app] 	at org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo.getMethod(ResteasyReactiveResourceInfo.java:65)
11:41:13,380 INFO  [app] 	at org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo.getParameterAnnotations(ResteasyReactiveResourceInfo.java:106)
11:41:13,381 INFO  [app] 	at org.jboss.resteasy.reactive.server.handlers.RequestDeserializeHandler.getAnnotations(RequestDeserializeHandler.java:115)
11:41:13,381 INFO  [app] 	at org.jboss.resteasy.reactive.server.handlers.RequestDeserializeHandler.isReadable(RequestDeserializeHandler.java:101)
11:41:13,381 INFO  [app] 	at org.jboss.resteasy.reactive.server.handlers.RequestDeserializeHandler.handle(RequestDeserializeHandler.java:62)
11:41:13,382 INFO  [app] 	at org.jboss.resteasy.reactive.server.handlers.RequestDeserializeHandler.handle(RequestDeserializeHandler.java:25)
11:41:13,382 INFO  [app] 	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:141)
11:41:13,382 INFO  [app] 	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:548)
11:41:13,383 INFO  [app] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
11:41:13,383 INFO  [app] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
11:41:13,383 INFO  [app] 	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
11:41:13,383 INFO  [app] 	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
11:41:13,384 INFO  [app] 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
11:41:13,384 INFO  [app] 	at java.lang.Thread.run(Thread.java:829)
11:41:13,385 INFO  [app] 	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:600)
11:41:13,385 INFO  [app] 	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)
11:41:13,385 INFO  [app] Caused by: java.lang.NoSuchMethodException: io.quarkus.ts.http.restclient.reactive.files.FileResource.upload(java.io.File)
11:41:13,386 INFO  [app] 	at java.lang.Class.getMethod(DynamicHub.java:1152)
11:41:13,386 INFO  [app] 	at org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo.getMethod(ResteasyReactiveResourceInfo.java:60)
11:41:13,386 INFO  [app] 	... 15 more
11:41:13,386 INFO  [app] 11:41:12,362 Request failed: java.lang.RuntimeException: java.lang.NoSuchMethodException: io.quarkus.ts.http.restclient.reactive.files.FileResource.upload(java.io.File)
11:41:13,387 INFO  [app] 	at org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo.getMethod(ResteasyReactiveResourceInfo.java:65)
11:41:13,387 INFO  [app] 	at org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo.getParameterAnnotations(ResteasyReactiveResourceInfo.java:106)
11:41:13,387 INFO  [app] 	at org.jboss.resteasy.reactive.server.handlers.RequestDeserializeHandler.getAnnotations(RequestDeserializeHandler.java:115)
11:41:13,388 INFO  [app] 	at org.jboss.resteasy.reactive.server.handlers.RequestDeserializeHandler.isReadable(RequestDeserializeHandler.java:101)
11:41:13,388 INFO  [app] 	at org.jboss.resteasy.reactive.server.handlers.RequestDeserializeHandler.handle(RequestDeserializeHandler.java:62)
11:41:13,388 INFO  [app] 	at org.jboss.resteasy.reactive.server.handlers.RequestDeserializeHandler.handle(RequestDeserializeHandler.java:25)
11:41:13,388 INFO  [app] 	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:141)
11:41:13,389 INFO  [app] 	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:548)
11:41:13,389 INFO  [app] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
11:41:13,389 INFO  [app] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
11:41:13,390 INFO  [app] 	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
11:41:13,390 INFO  [app] 	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
11:41:13,390 INFO  [app] 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
11:41:13,390 INFO  [app] 	at java.lang.Thread.run(Thread.java:829)
11:41:13,391 INFO  [app] 	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:600)
11:41:13,391 INFO  [app] 	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)
11:41:13,391 INFO  [app] Caused by: java.lang.NoSuchMethodException: io.quarkus.ts.http.restclient.reactive.files.FileResource.upload(java.io.File)
11:41:13,392 INFO  [app] 	at java.lang.Class.getMethod(DynamicHub.java:1152)
11:41:13,392 INFO  [app] 	at org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo.getMethod(ResteasyReactiveResourceInfo.java:60)
11:41:13,392 INFO  [app] 	... 15 more

```

### How to Reproduce?

1. `git clone git@github.com:fedinskiy/quarkus-test-suite.git -b reproducer/no-such-method`
2. `cd quarkus-test-suite/http/rest-client-reactive/`
3. `mvn clean verify -Dnative -Dit.test=LargeFileHandlingIT#uploadFileThroughClient -Dquarkus.platform.version=2.9.2.Final`. Any version, from 2.8.0.CR1 up to 999-SNAPSHOT will suffice and the test will fail.
Removing the `-Dnative` option, using version `2.7.6.Final` or running any other test will lead to successfull run.

### Output of `uname -a` or `ver`

5.17.11-300.fc36.x86_64

### Output of `java -version`

11.0.12, vendor: GraalVM Community

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.9.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)

### Additional information

_No response_

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java
@@ -6,6 +6,7 @@ import static io.quarkus.resteasy.reactive.common.deployment.QuarkusResteasyReac
 import static java.util.stream.Collectors.toList;
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.DATE_FORMAT;
 
+import java.io.File;
 import java.util.ArrayDeque;
 import java.util.ArrayList;
 import java.util.Collection;
@@ -182,6 +183,7 @@ public class ResteasyReactiveProcessor {
             DotName.createSimple(HttpServerRequest.class.getName()),
             DotName.createSimple(HttpServerResponse.class.getName()),
             DotName.createSimple(RoutingContext.class.getName()));
+    private static final DotName FILE = DotName.createSimple(File.class.getName());
 
     private static final int SECURITY_EXCEPTION_MAPPERS_PRIORITY = Priorities.USER + 1;
 
@@ -500,6 +502,10 @@ public class ResteasyReactiveProcessor {
                                                             QuarkusResteasyReactiveDotNames.IGNORE_METHOD_FOR_REFLECTION_PREDICATE)
                                                     .source(source)
                                                     .build());
+                                        }
+                                        if (parameterType.name().equals(FILE)) {
+                                            reflectiveClass.produce(new ReflectiveClassBuildItem(false, true, false,
+                                                    entry.getActualEndpointInfo().name().toString()));
                                         }
                                     }
                                 }

--- a/integration-tests/rest-client-reactive-multipart/src/main/java/io/quarkus/it/rest/client/multipart/MultipartClient.java
+++ b/integration-tests/rest-client-reactive-multipart/src/main/java/io/quarkus/it/rest/client/multipart/MultipartClient.java
@@ -21,6 +21,12 @@ import io.vertx.core.buffer.Buffer;
 public interface MultipartClient {
 
     @POST
+    @Path("/octet-stream")
+    @Consumes(MediaType.APPLICATION_OCTET_STREAM)
+    @Produces(MediaType.TEXT_PLAIN)
+    String octetStreamFile(File body);
+
+    @POST
     @Consumes(MediaType.MULTIPART_FORM_DATA)
     @Produces(MediaType.TEXT_PLAIN)
     @Path("/binary")

--- a/integration-tests/rest-client-reactive-multipart/src/main/java/io/quarkus/it/rest/client/multipart/MultipartResource.java
+++ b/integration-tests/rest-client-reactive-multipart/src/main/java/io/quarkus/it/rest/client/multipart/MultipartResource.java
@@ -53,6 +53,16 @@ public class MultipartResource {
     MultipartClient client;
 
     @GET
+    @Path("/client/octet-stream")
+    @Produces(MediaType.TEXT_PLAIN)
+    @Blocking
+    public String sendOctetStreamFile() throws IOException {
+        java.nio.file.Path tempFile = Files.createTempFile("dummy", ".txt");
+        Files.write(tempFile, "test".getBytes(UTF_8));
+        return client.octetStreamFile(tempFile.toFile());
+    }
+
+    @GET
     @Path("/client/byte-array-as-binary-file-with-pojo")
     @Consumes(MediaType.TEXT_PLAIN)
     @Produces(MediaType.TEXT_PLAIN)
@@ -226,6 +236,13 @@ public class MultipartResource {
         data.file = tempFile.toPath();
         data.number = NUMBER;
         return client.sendPathAsTextFile(data);
+    }
+
+    @POST
+    @Path("/echo/octet-stream")
+    @Consumes(MediaType.APPLICATION_OCTET_STREAM)
+    public String consumeOctetStream(File file) throws IOException {
+        return Files.readString(file.toPath());
     }
 
     @POST

--- a/integration-tests/rest-client-reactive-multipart/src/test/java/io/quarkus/it/rest/client/multipart/MultipartResourceTest.java
+++ b/integration-tests/rest-client-reactive-multipart/src/test/java/io/quarkus/it/rest/client/multipart/MultipartResourceTest.java
@@ -217,6 +217,18 @@ public class MultipartResourceTest {
         assertMultipartResponseContains(response, "file", MediaType.TEXT_PLAIN, HELLO_WORLD);
     }
 
+    @Test
+    public void shouldProperlyHandleOctetStreamFile() {
+        // @formatter:off
+        given()
+                .header("Content-Type", "text/plain")
+                .when().get("/client/octet-stream")
+                .then()
+                .statusCode(200)
+                .body(equalTo("test"));
+        // @formatter:on
+    }
+
     private void assertMultipartResponseContains(String response, String name, String contentType, Object value) {
         String[] lines = response.split("--");
         assertThat(lines).anyMatch(line -> line.contains(String.format(EXPECTED_CONTENT_DISPOSITION_PART, name))

